developers/weaviate/starter-guides/custom-vectors.mdx

```diff
@@ -8,187 +8,207 @@ image: og/docs/tutorials.jpg
+To install a client library, use the installer for the client language:
 
-import CreateSandboxCluster from '/_includes/wcs/create-cluster-sandbox.mdx'
+<Tabs groupId="languages">
+<TabItem value="py" label="Python Client v4">
+The v4 client requires Weaviate 1.23 or higher.<br/><br/>
 
-<CreateSandboxCluster />
+```bash
+pip install -U weaviate-client
+```
+</TabItem>
+<TabItem value="py3" label="Python Client v3">
 
-## Connect to your sandbox
+```bash
+pip install "weaviate-client==3.*"
+```
+</TabItem>
+<TabItem value="js" label="JS/TS Client v3">
 
-Follow these steps to connect to your sandbox instance.
+```bash
+npm install weaviate-client --tag next
+```
+</TabItem>
+<TabItem value="js2" label="JS/TS Client v2">
 
-### Get the connection details
+```bash
+npm install weaviate-ts-client
+```
+</TabItem>
+</Tabs>
 
-To connect to your sandbox instance, you need the following information:
+## Connect to Weaviate
 
-- The Weaviate sandbox URL
-- The Weaviate sandbox API key
+To connect to your Weaviate instance, you need the following information:
 
-To get the cluster URL and authentication details, follow these steps:
+- The Weaviate URL
+- The Weaviate API key
 
-1. Click the `Details` button to open the Details panel.
+If you are using a Sandbox instance, the URL and API keys are listed in the [details panel](/developers/wcs/quickstart#explore-the-details-panel) for your instance.
 
-import DetailsButton from '/developers/wcs/img/wcs-details-icon.jpg';
+To connect to Weaviate, run the code for your language to create a client object. Re-use the `client` object in the later steps to connect to your instance and run the sample code.
 
-<center><img src={DetailsButton} width="75%" alt="Details button"/></center>
+import ConnectToWeaviate from '/_includes/code/quickstart/connect.partial.mdx'
 
-2. Copy the REST endpoint.
+<ConnectToWeaviate />
 
-import RESTEndPoint from '/developers/wcs/img/wcs-rest-endpoint.jpg';
+## Collection definition
 
-<center><img src={RESTEndPoint} width="75%" alt="Details button"/></center>
+Weaviate stores [data](../concepts/data.md) in collections. Each data object in a collection has a set of properties and a vector representation. Before you import data, you should create a [collection definition](/developers/weaviate/config-refs/schema) to define the data properties for the collection.
 
-3. To get the API keys, click the `API keys` button. Copy the Weaviate API key.
+The data property definitions are called a schema. Although vectorizers are defined at the collection level, details, such as the specific model or behavior, are sometimes defined at the property level.
 
-import DetailsAPIKey from '/developers/wcs/img/wcs-console-details-api-key.jpg';
+This is a guide to using your own vectors. It assumes you already have vector embeddings for your data, so the sample configuration code sets the vectorizer to `none` on the collection level because the vector embeddings are included with the import file.
```

Comment:
I see that we set the vector distance metric on the PyV4 client. If that is intentional perhaps we could add a note that this is a configurable setting; which would be potentially important for users bringing their own vectors. 

I will separately leave comments on the individual code examples; but their settings seem different to each other and perhaps should be unifieid.